src/widgets/lv_img.c

```diff
@@ -693,17 +733,17 @@ static void draw_img(lv_event_t * e)
                     coords_tmp.x2 = coords_tmp.x1 + img->w - 1;
 
                     for(; coords_tmp.x1 < img_max_area.x2; coords_tmp.x1 += img_size_final.x, coords_tmp.x2 += img_size_final.x) {
-                        lv_draw_img(draw_ctx, &img_dsc, &coords_tmp, img->src);
+                        lv_draw_img(draw_ctx, &img_dsc, &coords_tmp, img->src.uri);
```

Comment:
Why no pass `img->src`? As it already contains the source type.

Reply:
That's where I put the limit between PR. The part 3 will modify the lv_draw_img signature to accept a `lv_img_src_t*` instead of void * so yes, that's exactly like you've thought.

lv_draw_img new version use the new image cache code and the new decoder code so I had to decide when to include it in the PR suite.

Reply:
Clear, and I agree to put the line here.